Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -547,7 +547,7 @@ For applications which require L2 protection, the MASVS states that: "The app in
 3. The application provides an overview of the last session after login at all times.
 4. The application has a self-service portal in which the user can see an audit-log and manage the different devices with which he can login.
 
-In all cases, the pentester should verify whether different devices are detected correctly. Therefore, the binding of the application to the actual device should be tested. For instance: in iOS a developer can use `identifierForVendor` whereas in Android, the developer can use `Settings.Secure.ANDROID_ID` to identify an application instance.(Note that starting at Android 8, `Android_ID` is no longer a device unique ID. Instead it becomes scoped by app-signing key, user and device. So checking `Android_ID` for device blocking could be tricky for these Android versions. Because if an app changes its signing key, the `Android_ID` will change and it won't be able to recognize old users devices) This together with keying material in the `Keychain` for iOS and in the `KeyStore` in Android can reassure strong device binding. Next, a pentester should test if using different IPs, different locations and/or different time-slots will trigger the right type of information in all scenarios.
+In all cases, the penetration tester should verify whether different devices are detected correctly. Therefore, the binding of the application to the actual device should be tested. For instance: in iOS a developer can use `identifierForVendor` whereas in Android, the developer can use `Settings.Secure.ANDROID_ID` to identify an application instance.(Note that starting at Android 8, `Android_ID` is no longer a device unique ID. Instead it becomes scoped by app-signing key, user and device. So checking `Android_ID` for device blocking could be tricky for these Android versions. Because if an app changes its signing key, the `Android_ID` will change and it won't be able to recognize old users devices) This together with keying material in the `Keychain` for iOS and in the `KeyStore` in Android can reassure strong device binding. Next, a penetration tester should test if using different IPs, different locations and/or different time-slots will trigger the right type of information in all scenarios.
```

Comment:
> application instance.(Note that starting at Android 8, 

to

> application instance. (Note that starting at Android 8,

Reply:
> So checking `Android_ID`

to

> So validating `Android_ID`

Reply:
> Instead it becomes scoped by app-signing key, user and device.

to

> Instead it becomes scoped by the combination of app-signing key, user and device.

Reply:
> Next, a penetration tester should test

to

> Next, you should test

Reply:
And maybe move the entire (Note) thing to a separate paragraph underneath the current one, as it's quite long for being in brackets ?

Reply:
Agree on all of it. Thanks